app/scripts/modules/core/src/pipeline/config/validation/anyFieldRequired.validator.ts

```diff
@@ -0,0 +1,47 @@
+import { module } from 'angular';
+import { map, some } from 'lodash';
+
+import { IPipeline, IStage, IStageOrTriggerTypeConfig, ITrigger } from 'core/domain';
+import { BaseRequiredFieldValidator, IRequiredField } from './baseRequiredField.validator';
+import { PIPELINE_CONFIG_VALIDATOR, PipelineConfigValidator } from './pipelineConfig.validator';
+import { IBaseRequiredFieldValidationConfig } from 'core/pipeline/config/validation/baseRequiredField.validator';
+
+export interface IMultiRequiredField extends IBaseRequiredFieldValidationConfig {
+  fields: IRequiredField[];
+}
+
+export type IAnyFieldRequiredValidationConfig = IBaseRequiredFieldValidationConfig & IMultiRequiredField;
+
+export class AnyFieldRequiredValidator extends BaseRequiredFieldValidator {
+  protected passesValidation(
+    pipeline: IPipeline,
+    stage: IStage | ITrigger,
+    validationConfig: IAnyFieldRequiredValidationConfig,
+  ): boolean {
+    return some(validationConfig.fields, (requiredField: IRequiredField) => {
+      return this.fieldIsValid(pipeline, stage, requiredField.fieldName);
+    });
+  }
+
+  protected validationMessage(
+    validationConfig: IAnyFieldRequiredValidationConfig,
+    config: IStageOrTriggerTypeConfig,
+  ): string {
+    let fieldString: string = map(validationConfig.fields, (requiredField: IRequiredField) =>
```

Comment:
Linter will be upset here. Also should use `Array.prototype.map` here. Lodash gives you null safety here but you should just do `(validationConfig.fields || []).map(...)` if that's what you're worried about.